hera_cal/abscal.py

```diff
@@ -981,510 +2636,319 @@ def TT_Phi_gain_arr(self):
+        nInt_to_load: number of integrations to load and calibrate simultaneously. Default -1 loads all integrations.
+        data_solar_horizon: Solar altitude threshold [degrees]. When the sun is too high in the data, flag the integration.
+        model_solar_horizon: Solar altitude threshold [degrees]. When the sun is too high in the model, flag the integration.
+        min_bl_cut: minimum baseline separation [meters] to keep in data when calibrating. None or 0 means no mininum,
+            which will include autocorrelations in the absolute calibration. Usually this is not desired, so the default is 1.0.
+        max_bl_cut: maximum baseline separation [meters] to keep in data when calibrating. None (default) means no maximum.
+        edge_cut: integer number of channels to exclude at each band edge in delay and global phase solvers
+        tol: baseline match tolerance in units of baseline vectors (e.g. meters)
+        phs_max_iter: integer maximum number of iterations of phase_slope_cal or TT_phs_cal allowed
+        phs_conv_crit: convergence criterion for updates to iterative phase calibration that compares them to all 1.0s.
+        clobber: if True, overwrites existing abscal calfits file at the output path
+        add_to_history: string to add to history of output abscal file
+
+    Returns:
+        hc: HERACal object which was written to disk. Matches the input redcal_file with an updated history and:
+            - gains: abscal gains for times that could be calibrated, redcal gains otherwise (but flagged)
+            - flags: redcal flags, with additional flagging if the data or model are flagged (see flag_utils.synthesize_ant_flags)
+            - quals: abscal chi^2 per antenna based on calibrated data minus model (Normalized by noise/nObs, but not with proper DoF)
+            - total_qual: abscal chi^2 based on calibrated data minus model (Normalized by noise/nObs, but not with proper DoF)
+    '''
+    # Raise error if output calfile already exists and clobber is False
+    if os.path.exists(redcal_file.replace(*output_replace)) and not clobber:
+        raise IOError("{} exists, not overwriting.".format(redcal_file.replace(*output_replace)))
+
+    # Load redcal calibration
+    hc = io.HERACal(redcal_file)
+    rc_gains, rc_flags, rc_quals, rc_tot_qual = hc.read()
+
+    # Initialize full-size, totally-flagged abscal gain/flag/etc. dictionaries
+    abscal_gains = copy.deepcopy(rc_gains)
+    abscal_flags = {ant: np.ones_like(rf) for ant, rf in rc_flags.items()}
+    abscal_chisq_per_ant = {ant: np.zeros_like(rq) for ant, rq in rc_quals.items()}
+    abscal_chisq = {pol: np.zeros_like(rtq) for pol, rtq in rc_tot_qual.items()}
+
+    # match times to narrow down model_files
+    matched_model_files = sorted(set(match_times(data_file, model_files, filetype='uvh5')))
+    if len(matched_model_files) > 0:
+        hd = io.HERAData(data_file)
```

Comment:
This gets a little tricky for non `uvh5` files b/c `pols` won't be loaded by default. We could do a hack and load a single integration, which would be quick and also load the relevant metadata if `filetype==miriad`